tests/rptest/tests/cloud_storage_chunk_read_path_test.py

```diff
@@ -185,7 +185,9 @@ def _assert_in_cache(self, expr: str):
     def test_read_chunks(self):
         self.default_chunk_size = 1048576
         self._set_params_and_start_redpanda(
-            cloud_storage_cache_chunk_size=self.default_chunk_size)
+            cloud_storage_cache_chunk_size=self.default_chunk_size,
+            # Disable leader balancer to have stable node to fetch metrics from.
+            enable_leader_balancer=False)
```

Comment:
would making the metric check be more flexible be an alternative?

Reply:
There will be always a chance of a race condition between querying who the leader is and the metric value.

So you have ideas? For this particular test it would probably be easier to just start one replica and not bother about leadership at all

Reply:
i approved the pr, i'm just asking if it's possible/reasonable.

> So you have ideas?

if in the face of leadership transfer there is a metrics query, potentially involving aggregation across endpoints, that even with the leadership transfer correctly asserts the property being tested for, then that would be the idea. not that it's easy or possible. just wondering, because less we assume about the system (e.g. balancer enabled/disabled) the more robust the tests are.